ansible/library/check_bgp_ipv6_routes_converged.py

```diff
@@ -0,0 +1,128 @@
+    if nh_diff_prefixes:
+        for prefix, expected, running in nh_diff_prefixes:
+            logging.warning(f"Prefix {prefix} nexthops not match, expected: {expected}, running: {running}")
+        return False
+
+    return True
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            expected_routes=dict(required=True, type='str'),
+            shutdown_ports=dict(required=True, type='list', elements='str'),
+            timeout=dict(required=False, type='int', default=300),
+            interval=dict(required=False, type='int', default=1),
+            log_path=dict(required=False, type='str', default='/tmp'),
+            compressed=dict(required=False, type='bool', default=False),
+            action=dict(required=False, type='str', choices=['shutdown', 'startup', 'no_action'], default='no_action')
+        ),
+        supports_check_mode=False
+    )
+    if module.params['log_path']:
+        config_module_logging("check_bgp_ipv6_routes_converged", log_path=module.params['log_path'])
+
+    logging.info("Start to check bgp routes converged at %s", datetime.datetime.now().strftime("%H:%M:%S"))
+
+    # check if need to decompress
+    if module.params.get('compressed', False):
+        # decompress
+        compressed_bytes = base64.b64decode(module.params['expected_routes'])
+        json_str = gzip.decompress(compressed_bytes).decode('utf-8')
+        expected_routes = json.loads(json_str)
+    else:
+        expected_routes = json.loads(module.params['expected_routes'])
+
+    shutdown_ports = module.params['shutdown_ports']
+    timeout = module.params['timeout']
+    interval = module.params['interval']
+    action = module.params.get('action', 'no_action')
+
+    # record start time
+    start_time = time.time()
+    logging.info("start time: %s", datetime.datetime.fromtimestamp(start_time).strftime("%H:%M:%S"))
+
+    # interface operation based on action
+    if action in ["shutdown", "startup"] and shutdown_ports:
+        ports_str = ",".join(shutdown_ports)
+        if action == "shutdown":
+            cmd = "sudo config interface shutdown {}".format(ports_str)
+        else:
+            cmd = "sudo config interface startup {}".format(ports_str)
+        logging.info("The command is: %s", cmd)
+        rc, out, err = module.run_command(cmd, executable='/bin/bash', use_unsafe_shell=True)
+        if rc != 0:
+            module.fail_json(msg=f"Failed to {action} ports: {err}")
+        logging.info("%s ports: %s", action, ports_str)
+    else:
+        logging.info("action is no_action or no shutdown_ports, skip interface operation.")
+
+    # check routes
+    check_count = 0
+    while True:
+        check_count += 1
+        logging.info(f"BGP routes check round: {check_count}")
+        running_routes = get_bgp_ipv6_routes(module)
+        if compare_routes(running_routes, expected_routes):
+            end_time = time.time()
```

Comment:
Instead of doing measuring end_time here, the time should be captured just before the get_bgp_ipv6_routes so it doesn't include vtysh command time delay.